lib/internal/http2/core.js

```diff
@@ -2038,21 +2043,24 @@ const setTimeout = {
   enumerable: true,
   writable: true,
   value: function(msecs, callback) {
-    if (typeof msecs !== 'number') {
-      throw new errors.TypeError('ERR_INVALID_ARG_TYPE',
-                                 'msecs',
-                                 'number');
-    }
+    // Type checking identical to timers.enroll()
+    msecs = validateTimerDuration(msecs);
+
     if (msecs === 0) {
-      unenroll(this);
+      clearTimeout(this[kTimeout]);
+
       if (callback !== undefined) {
         if (typeof callback !== 'function')
           throw new errors.TypeError('ERR_INVALID_CALLBACK');
         this.removeListener('timeout', callback);
       }
     } else {
-      enroll(this, msecs);
-      this[kUpdateTimer]();
+      clearTimeout(this[kTimeout]);
+      this[kTimeout] = setUnrefTimeout(() => {
+        this._onTimeout();
```

Comment:
This won't be feasible for `http` but for `http2`, I think we could move this off the publicly available `_onTimeout` and instead onto a Symbol.

Also, I would prefer if this was bound instead of using an arrow function but that's just a minor nit.

Reply:
Prefer that to be done in a separate commit. I'll switch to `bind()`, IIRC perf was good enough now?

Reply:
> Prefer that to be done in a separate commit.

I figured it could go here since this PR is what enables us to no longer use a publicly exposed method but either way works for me. 👍

Reply:
Hmm, true. Not sure how I feel about it.